### PR TITLE
Remove okhttp internal util usage

### PR DESCRIPTION
**Pull request by adrianha**
## Summary:

Remove okhttp internal `Util` usage so it can be compatible with [okhttp 5.0.0 alpha](https://square.github.io/okhttp/changelogs/changelog/#version-500-alpha7).

## Changelog:

[ANDROID] [CHANGED] - Remove okhttp3 internal util usage

## Test Plan:

None, the implementation is based on okhttp internal util

### CODE REVIEW

**Comment by javache** [comment]
Do we know if this has the same behaviour as before, where checked exceptions are ignored? https://github.com/square/okhttp/blob/8c8c3dbcfa91e28de2e13975ec414e07f153fde4/okhttp/src/commonMain/kotlin/okhttp3/internal/-UtilCommon.kt#LL282C1-L288C2

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java [hunk 1]
```diff
@@ -152,7 +151,7 @@ public void writeTo(BufferedSink sink) throws IOException {
           source = Okio.source(inputStream);
           sink.writeAll(source);
         } finally {
-          Util.closeQuietly(source);
+          source.close();
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:154]: Can we copy over the code of `closeQuitely`, namely:
```kotlin
fun Closeable.closeQuietly() {
  try {
    close()
  } catch (rethrown: RuntimeException) {
    throw rethrown
  } catch (_: Exception) {
  }
}
```

**Review by cortinico — CHANGES_REQUESTED** [review]

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java (hunk 1 above)
Comment (adrianha) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java:154]: thanks for the suggestion, i’ll update it later

**Comment by adrianha** [comment]
Source exception

just added the `IOException` catch block here, I think it should be enough to silence the exception as `Source` only throws `IOException`. Any thoughts @javache @cortinico?

**Comment by cortinico** [comment]
> just added the `IOException` catch block here, I think it should be enough to silence the exception as `Source` only throws `IOException`. Any thoughts @javache @cortinico?

I would suggest you just copy over the same code as `closeQuietly` (I would also add a comment to where the code was vendored from).

You're right that close only fires `IOException` but, as users might swap different Java & OkHTTP version, the exception types could differ in the future.

**Review by cortinico — CHANGES_REQUESTED** [review]
See my previous comment

**Review by cortinico — APPROVED** [review]

**Comment by adrianha** [comment]
thank you for reviewing @cortinico, any actionables from my side?

**Comment by cortinico** [comment]
Not for now, we'll be merging this internally. It can take up to a couple of days

**Pull request closed without merging**